lints/lint_mp_allowed_rsa_keys.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/********************************************************************
+Section 5.1 - Algorithms
+RSA keys whose modulus size in bits is divisible by 8, and is at least 2048.
+Section 5.2 - Forbidden and Required Practices
+CAs MUST NOT issue certificates that have:
+- invalid public keys (e.g., RSA certificates with public exponent equal to 1);
+********************************************************************/
+
+package lints
+
+import (
+	"crypto/rsa"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type allowedRSAKey struct{}
+
+func (l *allowedRSAKey) Initialize() error {
+	return nil
+}
+
+func (l *allowedRSAKey) CheckApplies(c *x509.Certificate) bool {
+	switch c.SignatureAlgorithm {
+	case x509.SHA1WithRSA, x509.SHA256WithRSA, x509.SHA384WithRSA, x509.SHA512WithRSA, x509.SHA256WithRSAPSS, x509.SHA384WithRSAPSS, x509.SHA512WithRSAPSS:
+		return true
+	}
+
+	return false
+}
+
+func (l *allowedRSAKey) Execute(c *x509.Certificate) *LintResult {
+	pubKey := c.PublicKey.(*rsa.PublicKey)
```

Comment:
For safety's sake I think it makes sense to handle a failed cast of the `c.PublicKey` as a `Fatal` level `LintResult`. In theory `CheckApplies` will prevent `Execute` from running into this but it defense in depth seems like the right call to me.
```suggestion
	pubKey, ok := c.PublicKey.(*rsa.PublicKey)
	if !ok  {
	  return &LintResult{Status: Fatal, Details: "certificate public key was not an RSA public key"}
	}
```

Reply:
I feel a little bit vindicated asking for this seeing that [CI has failed this branch](https://travis-ci.org/zmap/zlint/builds/592504764) because of a panic from this cast failing during the new integration tests:
```
2019/10/02 11:11:32 Reading data from "../data/xaa.csv"
panic: interface conversion: interface {} is *x509.AugmentedECDSA, not *rsa.PublicKey
goroutine 26 [running]:
github.com/zmap/zlint/lints.(*allowedRSAKey).Execute(0xbe27a8, 0xc000578400, 0xc000225701)
	/home/travis/gopath/src/github.com/zmap/zlint/lints/lint_mp_allowed_rsa_keys.go:48 +0x11e
github.com/zmap/zlint/lints.(*Lint).Execute(0xc00021f620, 0xc000578400, 0x849e39)
	/home/travis/gopath/src/github.com/zmap/zlint/lints/base.go:117 +0x9e
github.com/zmap/zlint.(*ResultSet).execute(0xc000dee420, 0xc000578400)
	/home/travis/gopath/src/github.com/zmap/zlint/zlint.go:44 +0x112
github.com/zmap/zlint.LintCertificate(0xc000578400, 0xc000d85650)
	/home/travis/gopath/src/github.com/zmap/zlint/zlint.go:83 +0x5c
github.com/zmap/zlint/integration.lintCertificate(0xc009a654af, 0x40, 0xc000578400, 0xc009bfb777, 0x40, 0x1000000)
	/home/travis/gopath/src/github.com/zmap/zlint/integration/corpus_test.go:23 +0x40
github.com/zmap/zlint/integration.TestCorpus.func2(0xc0000b2240, 0xc0000b2300, 0xc0096a6c10)
	/home/travis/gopath/src/github.com/zmap/zlint/integration/corpus_test.go:71 +0x6a
created by github.com/zmap/zlint/integration.TestCorpus
	/home/travis/gopath/src/github.com/zmap/zlint/integration/corpus_test.go:68 +0x1c6
FAIL	github.com/zmap/zlint/integration	1.803s
FAIL
```

Reply:
The test failed because of the SignatureAlgorithm check. The new check will be implemented using the code you provided, and so it won't be needed at Execute().